docs/04.guides/04.cookbooks/29.Threads-Scope/page.md

```diff
@@ -0,0 +1,272 @@
+		sleep(1000);
+	}
+	,true
+);
+dump("done in #getTickCount()-start#ms");
+</cfscript>
+```
+
+```each``` function has a option parallel, when we set parallel attribute as true. It runs each element parallel. In default it should be false.
+
+So that two array elements runs parallel done in 1second.
+
+
+```lucee
+<cfscript>
+tasks = [];
+loop from=1 to=100 index="i" {
+	arrayAppend(tasks,"t"&i)
+}
+start=getTickCount();
+tasks.each(
+	function(value, index, array){
+		systemOutput(value,true);
+		sleep(1000);
+	}
+	,true
+	//,100
+);
+dump("done in #getTickCount()-start#ms");
+</cfscript>
+```
+
+Here we have 100 array elements, setting option parallel as true it execute for 5 seconds not at 1 second because in default lucee runs 20 threads at the time after that it wait for free threads so only it take 5 seconds to complete.
+
+But we have option to set maximum threads open at that time. If it set to 100, it can open 100 threads at that time run parallel it take 1 second to complete.
+
+
+```each``` function is used not only for array, it used for struct & query also. Below example explain all the types.
+
+
+```lucee
+<cfscript>
+// ARRAY
+tasks = ["a","b"];
+tasks.each(
+	function(value, key, struct){
+		systemOutput(arguments,true);
+	}
+	,true
+);
+arrayEach(
+	tasks
+	,function(value, key, struct){
+		systemOutput(arguments,true);
+	}
+	,true
+);
+// STRUCT
+tasks = {a:1,b:2};
+structEach(
+	tasks
+	,function(value, key, struct){
+		systemOutput(arguments,true);
+	}
+	,true
+);
+tasks.each(
+	function(value, key, struct){
+		systemOutput(arguments,true);
+	}
+	,true
+);
+// QUERY
+persons = query(
+	'firstName':['Susi','Harry']
+	,'lastName':['Sorglos','Hirsch']
+);
+queryEach(
+	persons
+	,function(value, row, query){
+		systemOutput(arguments,true);
+	}
+	,true
+);
+persons.each(
+	function(value, row, query){
+		systemOutput(arguments,true);
+	}
+	,true
+);
+dump("done");
+</cfscript>
+```
+
+### Example 4 ###
+
+We are discussing for a long time how to extend functionality to make it easier to use or other new functionality.
+
+Here we explained about the future implementation of thread in lucee
+
+
+```lucee
+<cfscript>
+// Thread Pool
+tasks.each(
+	function(value, key, struct){
+		systemOutput(arguments,true);
+	}
+	,true
+	,20 // ATM default for max threads is 20, instead we plan to use a smart thread pool in the future (Java ExecutorService) 
+);
+</cfscript>
+```
+ATM we are using default max threads is 20, instead we plan to use a smart thread pool in the future it's based on your JVM(Java ExecutorService). So you don't want to take care how many threads were using system will do that and give best choice for you.
+
+
+```lucee
+<cfscript>
+thread /* action="thread" name="whatever" */ {
+	sleep();
+}
+</cfscript>
+```
+
+make thread smarter by also use a pool for threads.
+
+
+That the something we are planning it will come without changes in functionality on backend. That changes nothing in front end.
```

Comment:
***ToChange***
***This comment is for lines 246 through 249***
In the future we will make threads smarter by also using a pool for threads.

This feature is something we are planning. Changes will be implemented on the backend so that nothing changes on the front end.